src/charm.py

```diff
@@ -104,6 +106,7 @@ def _on_leader_elected(self, event) -> None:
             self._generate_passwords()
 
         self._update_hosts(event)
+        self.update_app_relation_data(check_mongo_users=False)
```

Comment:
maybe something like this will help

`MEDIAN_REELECTION_TIME=12`

```suggestion
        time.sleep(MEDIAN_REELECTION_TIME*2)
        self.update_app_relation_data(check_mongo_users=True)
```

Reply:
Thanks! Done. This works, just that the notion of sleep makes the charm integration tests a little flaky for CI gates.